src/_pytest/pytester.py

```diff
@@ -1061,7 +1065,27 @@ def run(self, *cmdargs):
             popen = self.popen(
                 cmdargs, stdout=f1, stderr=f2, close_fds=(sys.platform != "win32")
             )
-            ret = popen.wait()
+            timeout = kwargs.get('timeout')
+            if timeout is None:
+                ret = popen.wait()
+            elif six.PY3:
+                try:
+                    ret = popen.wait(timeout)
+                except subprocess.TimeoutExpired:
+                    raise self.TimeoutExpired()
+            else:
+                end = monotonic.monotonic() + timeout
```

Comment:
I don't think we need to introduce a new dependency to get an accurate timeout here, TBH. I think using `time.time()` on Linux and `time.clock()` on Windows will give us good enough accuracy? To be clear I like having the two implementations separate like this, makes it easier to drop the Python 2-only code later, I'm questioning the external dependency to achieve that.